src/menu/Menu.tsx

```diff
@@ -0,0 +1,274 @@
+import { create, tsx, renderer } from '@dojo/framework/core/vdom';
+import { createICacheMiddleware } from '@dojo/framework/core/middleware/icache';
+import { theme } from '@dojo/framework/core/middleware/theme';
+import { cache } from '@dojo/framework/core/middleware/cache';
+import { focus } from '@dojo/framework/core/middleware/focus';
+import { Keys } from '../common/util';
+import * as css from '../theme/menu.m.css';
+import MenuItem from './MenuItem';
+import { dimensions } from '@dojo/framework/core/middleware/dimensions';
+import global from '@dojo/framework/shim/global';
+import { RenderResult } from '@dojo/framework/core/interfaces';
+import { DimensionResults } from '@dojo/framework/core/meta/Dimensions';
+import { findIndex } from '@dojo/framework/shim/array';
+
+export type MenuOption = { value: string; label?: string; disabled?: boolean };
+
+interface MenuProperties {
+	/** Options to display within the menu */
+	options: MenuOption[];
+	/** The initial selected value */
+	initialValue?: string;
+	/** Callback called when user selects a value */
+	onValue(value: string): void;
+	/** Called to request that the menu be closed */
+	onRequestClose?(): void;
+	/** Optional callback, when passed, the widget will no longer control it's own active index / keyboard navigation */
+	onActiveIndexChange?(index: number): void;
+	/** Optional proprty to set the activeIndex when it is being controlled externally */
+	activeIndex?: number;
+	/** Determines if the widget can be focused or not. If the active index is controlled from elsewhere you may wish to stop the menu being focused and receiving keyboard events */
+	focusable?: boolean;
+	/** Callback called when menu root is focused */
+	onFocus?(): void;
+	/** Callback called when menu root is blurred */
+	onBlur?(): void;
+	/** Property to determine how many items to render. Not passing a number will render all results */
+	numberInView?: number;
```

Comment:
What do you think about `itemsInView` instead of `numberInView`?

Reply:
fine by me